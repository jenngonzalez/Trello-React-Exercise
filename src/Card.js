//The Card component should render markup matching the design.html: a div with a class of 'Card' containing an h3 for the card's title and a p element for the card's content.
// The Card component accepts 2 props: title and content.
// title is a string of the card's title.
// content is a string of the card's content.
// These props will be passed in for each Card from the List component.

import React from 'react';
import './Card.css';
// import { parentPort } from 'worker_threads'; -- what is this and where the heck did it come from????


function Card(props) {
    return (
        <div className="card">
            <button
            type='button'
            >
            delete
            </button>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
        </div>
    )
}


export default Card;
