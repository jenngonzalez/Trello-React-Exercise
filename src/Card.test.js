import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

describe ('Card component', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Card title="New Card" content="Lorem ipsum"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});

// it('renders the UI as expected when no messages', () => {
//     const tree = renderer
//     .create(<Messages name="Messages" unread={0}/>)
//     .toJSON();
//     expect(tree).toMatchSnapshot();
//     });
// });