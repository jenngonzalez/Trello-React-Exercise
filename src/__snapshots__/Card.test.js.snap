// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    New Card
  </h3>
  <p>
    Lorem ipsum
  </p>
</div>
`;
